From 6e075b47435661e41d48f91d19b91de0bdf6ef42 Mon Sep 17 00:00:00 2001
From: Brandon Williams <turbine.bmw@gmail.com>
Date: Sun, 26 Jul 2026 10:35:25 -0400
Subject: [PATCH 2138/2165] arm64: dts: qcom: x1e80100: drop QCE sleep
 bandwidth vote

The QCE driver installs a constant 393600 kB/s memory interconnect vote
at probe time. The X1E80100 device tree tags that path for active, wake
and sleep buckets, so the vote remains in the RPMh sleep batch after QCE
and the rest of the devices have suspended.

QCE cannot issue transactions after APSS enters suspend. Tag its memory
path active-only so the normal and wake constraints are preserved while
the sleep bucket can fall to zero.

Signed-off-by: Brandon Williams <turbine.bmw@gmail.com>
(cherry picked from commit 528279b54bf0b8a55f4cb24f0267aee9c54bdb1e)

diff --git a/arch/arm64/boot/dts/qcom/hamoa.dtsi b/arch/arm64/boot/dts/qcom/hamoa.dtsi
index 20939c9d69e6596f5eef6161d8aff9946afe292c..5d7b80ae57c40fa86482deb2a767ac333525608a 100644
--- a/arch/arm64/boot/dts/qcom/hamoa.dtsi
+++ b/arch/arm64/boot/dts/qcom/hamoa.dtsi
@@ -4026,8 +4026,8 @@ crypto: crypto@1dfa000 {
 				    "tx";
 			iommus = <&apps_smmu 0x480 0x0>,
 				 <&apps_smmu 0x481 0x0>;
-			interconnects = <&aggre2_noc MASTER_CRYPTO QCOM_ICC_TAG_ALWAYS
-					 &mc_virt SLAVE_EBI1 QCOM_ICC_TAG_ALWAYS>;
+			interconnects = <&aggre2_noc MASTER_CRYPTO QCOM_ICC_TAG_ACTIVE_ONLY
+					 &mc_virt SLAVE_EBI1 QCOM_ICC_TAG_ACTIVE_ONLY>;
 			interconnect-names = "memory";
 		};
 
